hosts/utils_test.go

```diff
@@ -42,3 +46,67 @@ func TestOnlineChecker(t *testing.T) {
 		t.Fatal("expected")
 	}
 }
+
+// mockResolverFallback is for testing the behavior of the default Resolver and
+// making sure the error handling and fallback works correctly.
+type mockResolverFallback struct {
+	addrs []net.IPAddr
+	err   error
+}
+
+func (m *mockResolverFallback) LookupIPAddr(ctx context.Context, host string) ([]net.IPAddr, error) {
+	return m.addrs, m.err
+}
+
+func TestResolver_LookupIPAddr(t *testing.T) {
```

Comment:
```suggestion
func TestResolver(t *testing.T) {
```

That should be fine. No need to introduce a new naming convention for tests.